src/asm/asm.s

```diff
@@ -334,3 +334,15 @@ _x86_64_asm_xsetbv:
     movl  %esi, %eax    # Second param is the low 32-bits
     xsetbv              # Third param (high 32-bits) is already in %edx
     retq
+
+.global _x86_64_asm_write_mxcsr
+.p2align 4
+_x86_64_asm_write_mxcsr:
+    ldmxcsr %rdi
```

Comment:
In hindsight this wasn't as straight forward as I thought, `ldmxcsr` expects a memory operand. `%rdi` should be put into parentheses and the corresponding external function  declaration in rust should take a pointer to `MxCsr`. 

Note that this is also what the rust code using inline assembly does: it passes a reference.

We do something similar when executing `lgdt`: https://github.com/rust-osdev/x86_64/blob/08b520ef770e55538e65db11d46932305089b71a/src/asm/asm.s#L117 And
https://github.com/rust-osdev/x86_64/blob/08b520ef770e55538e65db11d46932305089b71a/src/asm/mod.rs#L163